src/StoreMenu/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { color, space, borderRadius } from '@core-ds/primitives';
+import * as React from 'react';
+import styled from 'styled-components';
+import { Menu, MenuButton, MenuList, MenuItem } from '../Menu';
+import { StoreFlag } from './StoreFlag';
+
+export interface StoreMenuProps {
+   className?: string;
+   label: string;
+   initialCountryCode?: string;
+   stores: Store[];
+   selectedCountryCode?: string;
+   onChange?: (store: Store) => void;
+}
+
+interface Store {
+   name: string;
+   countryCode: string;
+   currency: string;
+   url?: string;
+}
+
+function UnstyledStoreMenu({
+   className,
+   label,
+   initialCountryCode,
+   stores,
+   selectedCountryCode,
+   onChange,
+}: StoreMenuProps) {
+   const [internalCountryCode, setInternalCountryCode] = React.useState(initialCountryCode);
+
+   const countryCode = React.useMemo(() => {
+      return selectedCountryCode || internalCountryCode;
+   }, [selectedCountryCode, internalCountryCode]);
```

Comment:
This seems like a net performance loss over just `const countryCode = selectedCountryCode || internalCountryCode;`

Reply:
Agree, here `useMemo` is totally useless 👍 
```suggestion
   const countryCode = selectedCountryCode || internalCountryCode;
```